src/bindings/python/src/openvino/frontend/jax/jaxpr_decoder.py

```diff
@@ -0,0 +1,310 @@
+        Inputs: 
+            - jaxpr: for users, `ClosedJaxpr` is expected here. See https://github.com/google/jax/blob/jaxlib-v0.4.29/jax/_src/core.py#L197
+            - name: the name for the model.
+            - literals: the literals (constants) that are used in the model.
+        '''
+        Decoder.__init__(self)
+        
+        if isinstance(jaxpr, (jax.core.JaxprEqn, jax.core.Jaxpr, jax.core.Var)):
+            self.jaxpr = jaxpr
+        elif isinstance(jaxpr, jax.core.ClosedJaxpr):
+            # Take the `Jaxpr` from `ClosedJaxpr`, see https://github.com/google/jax/blob/jaxlib-v0.4.29/jax/_src/core.py#L85
+            self.jaxpr = jaxpr.jaxpr
+            # Literal should be a `Jax.core.Var`, see https://github.com/google/jax/blob/jaxlib-v0.4.29/jax/_src/core.py#L85
+            self.literals = jaxpr.literals
+        else:
+            raise ValueError(f"Unexpected type of jaxpr: {type(jaxpr)}")
+        self.name = name
+        if self.name is None:
+            self.name = "jax_module"
+        if literals is not None:
+            self.literals = literals
+            
+        self.params = {}
+        if hasattr(self.jaxpr, 'params') and isinstance(self.jaxpr.params, dict):
+            for k in self.jaxpr.params.keys():
+                self.params[k] = self.convert_param_to_constant_node(self.jaxpr, k)
+                
+        # TODO: this implementation may lead to memory increasing. Any better solution?
+        self.m_decoders = []
+        
+    def inputs(self) -> List[int]:
+        if isinstance(self.jaxpr, jax.core.Var):
+            return []
+        elif isinstance(self.jaxpr, jax.core.JaxprEqn):
+            idx = 0
+            res = []
+            for inp in self.jaxpr.invars:
+                if isinstance(inp, jax.core.Literal):
+                    res.append(self.literals[idx].output(0))
+                    idx += 1
+                else:
+                    res.append(id(inp))
+            return res
+        else:
+            return [id(v) for v in self.jaxpr.invars]
+    
+    def input(self, idx: int) -> int:
+        if isinstance(self.jaxpr, jax.core.Var):
+            raise IndexError("The jaxpr is a constant, which does not have input.")
+        else:
+            return id(self.jaxpr.invars[idx])
+    
+    def get_input_shape(self, index):
+        if isinstance(self.jaxpr, jax.core.Var):
+            raise IndexError("The jaxpr is a constant, which does not have input shape.")
+        else:
+            return PartialShape(self.jaxpr.invars[index].aval.shape)
+    
+    def get_input_signature_name(self, index) -> str:
+        return "jaxpr_invar_" + str(index)
+    
+    def get_input_type(self, index) -> OVType:
+        if isinstance(self.jaxpr, jax.core.Var):
+            raise IndexError("The jaxpr is a constant, which does not have input type.")
+        else:
+            return get_ov_type_for_value(self.jaxpr.invars[index])
+        
+    def get_named_param(self, name):
```

Comment:
please give description for JAX FE methods (that are not a part of common FE API)